python/setup.py

```diff
@@ -264,11 +264,8 @@ def get_packages(self):
 
     setup_spec.extras["rllib"] = setup_spec.extras["tune"] + [
         "dm_tree",
-        "gym>=0.21.0,<0.24.0",
+        "gymnasium==0.26.3",
```

Comment:
This does not reflect the [specification](https://github.com/ray-project/ray/blame/master/python/requirements/ml/requirements_rllib.txt#L7-L8) in `python/requirements/ml/requirements_rllib.txt`

Those extras themselves cause a whole cascade of dependencies, and so this should be fully specified in `setup.py`.

Also note that the `mujoco-py<2.2,>=2.1` pin [there](https://github.com/ray-project/ray/blame/master/python/requirements/ml/requirements_rllib.txt#L9-L10) does not correspond to `gymnasium[mujoco]` - the former is the original closed source versions, the latter includes a different `mujoco` (after the opensourcing), c.f. [here](https://github.com/Farama-Foundation/Gymnasium/blame/v0.26.3/setup.py#L40-L41)